LPS-105855 make changes needed to release Data Engine as part of core

leoadb opened the pull request.

Commit:
LPS-105855 copy from liferay-portal/7.2.x hash ded0e9390637985231e962e4ad4cfa4639eabb26

Commit:
LPS-105855 export packages

Commit:
LPS-105855 move to v1_0 package. “storage” package should be inside a “version” package because we reference some versioned resources.

Commit:
LPS-105855 move to v2_0 package, same reason of previous commit

Commit:
LPS-105855 add property so we can filter the correct implementation. I’ve left the current version with no property so someone might add an implementation without the need of explicitly define a version

Commit:
LPS-105855 had to applied these changes due to source formatter changes

Commit:
LPS-105855 make changes needed to release Data Engine as part of core

liferay-continuous-integration commented:
CI is automatically triggering &quot;ci:test:sf&quot; and &quot;ci:test:relevant&quot; for this pull to run Source Formatter and relevant tests.

Comment &quot;ci:test&quot; to run the full PR Tester for this pull.

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest(master)/6476/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-3.liferay.com//userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6476/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-3.liferay.com/job/test-portal-source-format/2931/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-3.liferay.com//userContent/jobs/test-portal-source-format/builds/2931/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 2 minutes 46 seconds 365 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/cb34aedc8a1b7fc01bada03ccf023a13e2712ac6">cb34aedc8a1b7fc01bada03ccf023a13e2712ac6</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/leoadb/liferay-portal/tree/LPS-105855">LPS-105855</a><br/>Branch GIT ID: <a href="https://github.com/leoadb/liferay-portal/commit/d36dd760108766061f690148e77b6ea66ab238ec">d36dd760108766061f690148e77b6ea66ab238ec</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-3.liferay.com/job/test-portal-source-format/2931/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/2931/jenkins-report.html">here</a>.</h5></details></html>

leoadb commented:
@brianchandotcom second try, please see your comments in https://github.com/brianchandotcom/liferay-portal/pull/82065#issuecomment-566361022

My anwsers are below:

**I thought we were going to do this afterward.. like in March. Why did we do this now?**
I saw a Jason's post from November 26th in Slack saying we **(must or should)** send all breaking changes by Dec 31. Is this a **"must"** or a **"should"**? Are we going to be able to send breaking changes between GA releases? I've undertood that all breaking changes **must** be sent until GA1, so that's why I am sending this to you now. 

**Also, when you say, copy from 7.2.x, is that liferay-portal/7.2.x or liferay-portal-ee/7.2.x. (Ideally, you tell me the exact hash from 7.2.x that you got this from. I'm going to check that 7.2.x, and rsync it myself to make sure the copy was correct).**
I've copied from liferay-portal/7.2.x hash ded0e9390637985231e962e4ad4cfa4639eabb26. But a simple copy is not possible due to:
1- Source formatter changes as you can see in b91eb46422e74b09711f15680da261f9bf752a3f
2- I've moved some classes from package **com.liferay.data.engine.rest.internal.storage** to **com.liferay.data.engine.rest.internal.storage.v1_0** and **com.liferay.data.engine.rest.internal.storage.v2_0** as we were referencing some * *versioned* * resources. See commits from 2e1120d146b57983edfead07107a0935f207a014 to 8ded0688bbef2c4a3ebea92a22bdeb3a593051f7

I've updated commit messages to make myself clear.

Thanks a lot.

cc @jeyvison @jpince @brunofarache

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 8 out of 8 jobs passed</h3><h3>:x: ci:test:relevant - 19 out of 21 jobs passed in 1 hour 12 minutes 7 seconds 152 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/cb34aedc8a1b7fc01bada03ccf023a13e2712ac6">cb34aedc8a1b7fc01bada03ccf023a13e2712ac6</a></p><h4>Copied in Private Modules Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal-ee/tree/master-private">master-private</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal-ee/commit/6c0b5322d1e18eac4342d74f6522c7b7a9bbd7ed">6c0b5322d1e18eac4342d74f6522c7b7a9bbd7ed</a></p><details><summary><strong>ci:test:stable - 8 out of 8 jobs PASSED</strong></summary><details><summary><strong>8 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/344893/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305341/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/352304/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305340/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/344894/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/342809/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305337/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/364901/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li></ul></details></details><details><summary><strong>ci:test:relevant - 19 out of 21 jobs PASSED</strong></summary><div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest(master)/6476/"><strike><strong>test-portal-acceptance-pullrequest(master)</strong></strike></a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/364903/"><strike><strong>test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</strong></strike></a></li></ul></div><details><summary><strong>19 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/291972/">test-portal-acceptance-pullrequest-batch(master)/central-requirements-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/364902/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk11_open/0</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/344893/">test-portal-acceptance-pullrequest-batch(master)/functional-smoke-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305341/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/0</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/352304/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/1</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305340/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/2</a></li><li><a href="https://test-1-19.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/344894/">test-portal-acceptance-pullrequest-batch(master)/functional-upgrade-tomcat90-mysql57-jdk8/3</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/291970/">test-portal-acceptance-pullrequest-batch(master)/js-unit-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/345312/">test-portal-acceptance-pullrequest-batch(master)/js-unit-private-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/345313/">test-portal-acceptance-pullrequest-batch(master)/lpkg-base-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/345311/">test-portal-acceptance-pullrequest-batch(master)/lpkg-controller-jdk8</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/345314/">test-portal-acceptance-pullrequest-batch(master)/modules-compile-jdk8</a></li><li><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/342809/">test-portal-acceptance-pullrequest-batch(master)/modules-integration-mysql57-jdk8</a></li><li><a href="https://test-1-21.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/291971/">test-portal-acceptance-pullrequest-batch(master)/modules-semantic-versioning-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305337/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305338/">test-portal-acceptance-pullrequest-batch(master)/modules-unit-project-templates-jdk8</a></li><li><a href="https://test-1-23.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/211349/">test-portal-acceptance-pullrequest-batch(master)/pmd-jdk8</a></li><li><a href="https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/305339/">test-portal-acceptance-pullrequest-batch(master)/service-builder-jdk8</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/364901/">test-portal-acceptance-pullrequest-batch(master)/unit-jdk8</a></li></ul></details></details><h5>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6476/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/364903/">test-portal-acceptance-pullrequest-batch(master)/semantic-versioning-jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=0,label_exp=!master/364903//consoleText">AXIS_VARIABLE=0,label_exp=!master #364903</a><pre><code>     [exec] &gt; Task :dxp:apps:oauth:oauth-api:compileJava FAILED
     [exec] [ant:path] Note: Some input files use or override a deprecated API.
     [exec] An exception has occurred in the compiler (1.8.0_121). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.
     [exec] [ant:antelope:stringutil] java.lang.NullPointerException
     [exec] [ant:var] 	at org.apache.tools.ant.UnknownElement.handleErrorOutput(UnknownElement.java:263)
     [exec] [ant:var] 	at org.apache.tools.ant.Project.demuxOutput(Project.java:1279)
     [exec] [ant:var] 	at org.apache.tools.ant.DemuxOutputStream.processBuffer(DemuxOutputStream.java:170)
     [exec] [ant:var] 	at org.apache.tools.ant.DemuxOutputStream.write(DemuxOutputStream.java:145)
     [exec] [ant:var] 	at org.apache.tools.ant.DemuxOutputStream.write(DemuxOutputStream.java:239)
     [exec] [ant:var] 	at java.io.PrintStream.write(PrintStream.java:480)
     [exec] [ant:var] 	at sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:221)
     [exec] [ant:var] 	at sun.nio.cs.StreamEncoder.implFlushBuffer(StreamEncoder.java:291)
     [exec] [ant:var] 	at sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:295)
     [exec] [ant:var] 	at sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:141)
     [exec] [ant:var] 	at java.io.OutputStreamWriter.flush(OutputStreamWriter.java:229)
     [exec] [ant:var] 	at java.io.BufferedWriter.flush(BufferedWriter.java:254)
     [exec] [ant:var] 	at java.io.PrintWriter.newLine(PrintWriter.java:482)
     [exec] [ant:var] 	at java.io.PrintWriter.println(PrintWriter.java:629)
     [exec] [ant:var] 	at java.io.PrintWriter.println(PrintWriter.java:740)
     [exec] [ant:var] 	at com.sun.tools.javac.util.Log.printRawLines(Log.java:531)
     [exec] [ant:var] 	at com.sun.tools.javac.util.Log.writeDiagnostic(Log.java:622)
     [exec] [ant:var] 	at com.sun.tools.javac.util.Log$DefaultDiagnosticHandler.report(Log.java:584)
     [exec] [ant:var] 	at com.sun.tools.javac.util.Log.report(Log.java:562)
     [exec] [ant:var] 	at com.sun.tools.javac.util.AbstractLog.mandatoryNote(AbstractLog.java:239)
     [exec] [ant:if] 	at com.sun.tools.javac.util.MandatoryWarningHandler.logMandatoryNote(MandatoryWarningHandler.java:266)</code></pre></div></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream(master)">here</a>.</h4></details></html>

brianchandotcom commented:
@leoadb breaking changes do need to come before GA1.. but  adding v1_0 in March is NOT a breaking change.

Breaking changes are if we changed what v2_0 is.

Are we going to change ore remove exported packages? If so, send me those changes now. But there's no need to move v1_0 now.
